xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/webjar/editBus.js

```diff
@@ -0,0 +1,232 @@
+   * live data.
+   * This is useful to know when to save the table, and to know which cells can be edited according to the current edit
+   * state
+   */
+  class EditBusService {
+
+    /**
+     * Default constructor.
+     * @param editBus an edit bus Vue object
+     * @param logic the live data logic instance
+     */
+    constructor(editBus, logic) {
+      this.editBus = editBus;
+      this.editStates = {};
+      this.editStatesTimeouts = {};
+      this.logic = logic;
+    }
+
+    /**
+     * Initializes the Vue events listeners.
+     */
+    init() {
+      this.editBus.$on('start-editing-entry', ({entryId, propertyId}) => {
+        const entryState = this.editStates[entryId] || {};
+        const propertyState = entryState[propertyId] || {};
+        propertyState.editing = true;
+        entryState[propertyId] = propertyState;
+        this.editStates[entryId] = entryState;
+      })
+
+      this.editBus.$on('cancel-editing-entry', ({entryId, propertyId}) => {
+        const entryState = this.editStates[entryId];
+        const propertyState = entryState[propertyId];
+
+        // The entry is not edited anymore.
+        // The content is not edited, and should be `undefined` if the property was edited for the first time.
+        // If the property was edited and saved a first time, then edited and cancelled, the content must stay to one from
+        // the first edit.
+        propertyState.editing = false;
+
+      })
+
+      this.editBus.$on('save-editing-entry', ({entryId, propertyId, content}) => {
+        const entryState = this.editStates[entryId];
+        const propertyState = entryState[propertyId];
+        // The entry is not edited anymore but its content will need to be saved once the rest of the properties of the 
+        // entry are not in edit mode. 
+        propertyState.editing = false;
+        propertyState.tosave = true;
+        propertyState.content = content;
+        this.save(entryId);
+      })
+    }
+
+    /**
+     * Save the changed values of an entry server side.
+     * @param entryId the entry id of the entry to save
+     */
+    save(entryId) {
+      const values = this.editStates[entryId];
+      var canBeSaved = false;
+      var keyEntry = undefined;
+
+
+      // Look for the single cell to save.
+      for (keyEntry in values) {
+        const entryValue = values[keyEntry];
+
+        const editing = entryValue.editing;
+        const tosave = entryValue.tosave;
+        canBeSaved = !editing && tosave;
+
+        if (canBeSaved) break;
+      }
+
+      // If a cell to save is found, we aggegrate the content to save and save it. 
+      if (canBeSaved && keyEntry) {
+        // Aggregates the content of the form values of each property.
+
+        const vals = {};
+        values[keyEntry].content.forEach((content) => {
+          aggregate(content, vals);
+        })
```

Comment:
I find this overly complicated and the root cause is the format of "content" you pass to the same function. IMO you should either pass an plain object or an array of objects with the same structure (as in the case of ``serializeArray``).